protocol/river-layout-unstable-v1.xml

```diff
@@ -0,0 +1,205 @@
+    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+  </copyright>
+
+  <description summary="let clients propose view positions and dimensions">
+    This protocol specifies a way for clients to propose arbitrary positions and
+    dimensions for a set amount of views on a specific output of a compositor
+    through the river_layout object.
+
+    These views are represented as a simple list. Individual views are
+    identifiable by their list index. This index is at minimum 0, which
+    corresponds to the first view, and at maximum one less than the total amount
+    of views in the list, which corresponds to the last view.
+
+    The entire set of proposed positions and dimensions for the views in the
+    list are called a layout. Due to their list heritage, layouts are also
+    logically strictly linear; Any complex underlying data structure a client
+    may use when generating the layout is lost in transmission. This is an
+    intentional limitation.
+
+    An arbitrary amount of views starting from the top of the list are
+    designated as the so called main views. This is a hint to let clients know
+    which views are important to the user. The client could then for example
+    decide to place these windows at prominent positions. Additionally a size
+    factor hint is provided. Clients may choose to ignore these hints.
+
+    Note that the client may need to handle multiple layout demands per
+    river_layout object simultaneously.
+  </description>
+
+  <interface name="zriver_layout_manager_v1" version="1">
+    <description summary="manage river layout objects">
+      A global factory for river_layout objects.
+    </description>
+
+    <enum name="error">
+      <entry name="invalid_serial" value="0" summary="serial does not belong to an active layout demand">
+      <entry name="invalid_index" value="0" summary="index does not belong to a view in the list">
+    </enum>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the river_layout_manager object">
+        This request indicates that the client will not use the
+        river_layout_manager object any more. Objects that have been created
+        through this instance are not affected.
+      </description>
+    </request>
+
+    <request name="get_river_layout">
+      <description summary="create a river_layout object">
+        This creates a new river_layout object for the given wl_output.
+
+        All layout related communication is done through this interface.
+
+        The namespace is used by the compositor to decide which river_layout
+        object will receive layout demands for the output. Namespaces must be
+        globally unique. If the namespace is already in use by a different
+        river_layout object, this request will fail.
```

Comment:
Requests cannot fail in wayland aside from a protocol error which is fatal and terminates the client. We need to send an event on the `river_layout` object to handle this case:
```suggestion
        globally unique. If the namespace is already in use by a different
        river_layout object, the server will send a namspace_in_use event on the river_layout object immediately after creation and ignore all requests on the object aside from the destroy request.
```